src/component/layout/Nav.jsx

```diff
@@ -0,0 +1,57 @@
+import { useEffect } from "react";
+import { useSelector, useDispatch } from "react-redux";
+import { filterNews } from "../../features/NewsSlice";
+import { AiOutlineSearch } from "react-icons/ai";
+import NewsPublisher from "../ui/NewsPublisher";
+import { RiNewspaperLine } from "react-icons/ri";
+
+const Nav = () => {
+  const dispatch = useDispatch();
+  const { filter } = useSelector((state) => state.news);
+  const { showPublishers } = useSelector((state) => state.news);
```

Comment:
you can use destructuring to extract the  filter and showPublishers , in order to avoid repeating the  code 
```suggestion
  const { filter ,showPublishers} = useSelector((state) => state.news);

```